### DISCUSSION TITLE
How does hive-partitioning work with read_parquet?

### DISCUSSION THREAD

**Discussion by erezinman** [duckdb/duckdb#14164] [General]
Can anyone describe? So far, my guess (at least with connection over `fsspec` in python), is that first all the paths are globbed, and then they are parsed as columns. 

For example, in the query
```sql
SELECT FIRST(filename)
FROM read_parquet("path/to/root/col1=*/col2=*/**/*.parquet", filename=true)
WHERE (col1 = "a" and col2 = "b")
```
First, the glob pattern will be performed as a whole (obtaining values other than `(col1 = "a" and col2 = "b")`), and then these values will be filtered afterwards (potentially obtaining a lot more files than required).

This is in contrast to other libraries like `pyarrow` that require you to know the hive-schema beforehand, so it can traverse/walk only the filtered directories.

Does anyone know if this is indeed the case?

**Comment by soerenwolfers** [comment]
see https://github.com/duckdb/duckdb/discussions/7620
